dashboard/src/components/AppList/AppList.test.tsx

```diff
@@ -19,70 +20,82 @@ beforeEach(() => {
   };
 });
 
-it("renders a loading message if apps object is empty", () => {
-  const wrapper = shallow(<AppList {...defaultProps} />);
-  expect(wrapper.text()).toBe("Loading");
-});
+describe("while fetching apps", () => {
+  beforeEach(() => {
+    props = { ...defaultProps, apps: { isFetching: true } };
+  });
 
-it("renders a loading message if it's fetching apps", () => {
-  const wrapper = shallow(
-    <AppList
-      {...defaultProps}
-      apps={
-        {
-          isFetching: true,
-          items: [],
-          listOverview: [],
-          listingAll: false,
-        } as IAppState
-      }
-    />,
-  );
-  expect(wrapper.text()).toContain("Loading");
+  it("matches the snapshot", () => {
+    const wrapper = shallow(<AppList {...props} />);
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it("renders a loading message", () => {
+    const wrapper = shallow(<AppList {...props} />);
+    expect(wrapper.text()).toContain("Loading");
+  });
+
+  it("renders a Application header", () => {
+    const wrapper = shallow(<AppList {...props} />);
+    expect(wrapper.find("h1").text()).toContain("Applications");
+  });
+
+  it("does not show the search filter nor deploy button", () => {
+    const wrapper = shallow(<AppList {...props} />);
+    expect(wrapper.find("SearchFilter")).not.toExist();
+    expect(wrapper.find("label.checkbox")).not.toExist();
+    expect(wrapper.find(".deploy-button")).not.toExist();
+  });
 });
 
-it("renders a welcome message if no apps are available", () => {
-  const wrapper = shallow(
-    <AppList
-      {...defaultProps}
-      apps={
-        {
-          isFetching: false,
-          items: [],
-          listOverview: [],
-          listingAll: false,
-        } as IAppState
-      }
-    />,
-  );
-  expect(wrapper).toMatchSnapshot();
+describe("when fetched but not apps available", () => {
+  beforeEach(() => {
+    props = { ...defaultProps, apps: { listOverview: [] } };
+  });
+
+  it("matches the snapshot", () => {
+    const wrapper = shallow(<AppList {...props} />);
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it("renders a welcome message", () => {
+    const wrapper = shallow(<AppList {...props} />);
+    expect(wrapper.find("MessageAlertPage")).toExist();
+  });
+
+  it("shows the search filter, show deleted checkbox and deploy button", () => {
```

Comment:
this test is the same one than `it("renders the 'Show deleted apps' button even if the app list is empty"`. Please remove that one to avoid duplicity.

Reply:
did you miss this comment?

Reply:
They are the same test but in different context no?

One of them in `when fetched but not apps available` and the other one under `while fetching apps` (note the describe and beforeAll callbacks)

Reply:
yeah, it's a different scenario, I actually thought @andresmgot comment was for an earlier duplication that you had fixed

Reply:
Ahh cool, thanks for pointing it out.